tests/test_standard_input_json.py

```diff
@@ -1,23 +1,169 @@
+        """
+        Test that a Gen3 JSON format input file can be uploaded to the DSS,
+        and that all of the data files loaded are loaded by reference
+        and set to not be indexed.
+
+        1. Generates a Gen3 JSON input file from a template with a new unique 'bundle_did'
+           and a new 'did' for the first file in the bundle.
+        2. Searches the DSS to make sure it doesn't already exist using the HCA CLI Python bindings.
+        3. Uploads the gen3 json to the DSS.
+        4. Searches the DSS to verify the bundle was uploaded and indexed successfully.
+        5. Assert data files are loaded by reference and set to not be indexed.
+        6. Assert that the new 'did' for the first file in the bundle was found in the results.
+        """
+
+        message("Test that initial loading works successfully")
+        # mint a new 'bundle_did'
+        guid = str(uuid.uuid4())
+        # make new guid for first file
+        file_guid = f'dg.4503/{str(uuid.uuid4())}'
+        # we want a new version of the file to be uploaded
+        file_version = datetime.datetime.now(datetime.timezone.utc).isoformat()
+        self._test_gen3_loading(test_json, guid, file_guid, file_version)
+
+        message("Test that uploading again will be handled successfully")
+        guid = str(uuid.uuid4())
+        self._test_gen3_loading(test_json, guid, file_guid, file_version)
+
+    def _test_gen3_loading(self, test_json, bundle_guid, file_guid, file_version):
+
+        @eventually(timeout_seconds=5.0, retry_interval_seconds=1.0)
+        def _search_for_bundle(bundle_uuid):
+            # Search for the bundle uuid in the DSS and make sure it now exists and uploading was successful
+            search_results = self.dss_client.post_search(es_query={'query': {'term': {'uuid': bundle_uuid}}}, replica='aws')
+            assert search_results['total_hits'] > 0
+            return search_results
+
+        message("Search for the bundle uuid in the DSS to make sure it does not exist yet")
+        search_results = self.dss_client.post_search(es_query={'query': {'term': {'uuid': bundle_guid}}}, replica='aws')
+        assert search_results['total_hits'] == 0
+
+        message("Prepare test input file to load")
+        with self._tmp_json_file(test_json, bundle_guid, file_guid, file_version) as tmp_json:
+            message("Load the test input file")
+            self._load_file(tmp_json)
+
+            message("Wait for newly loaded bundle to appear in search results")
+            search_results = _search_for_bundle(bundle_guid)
+
+            message("Verify that all of the results (except metadata.json) are file references "
+                    "and set to not be indexed")
+            found_matching_file = False
+            for r in search_results['results']:
+                response = requests.get(r['bundle_url'])
+                returned_json = response.json()
+                for f in returned_json['bundle']['files']:
```

Comment:
This test really should verify that the number of data files loaded into the bundle matches those specified in the input file. As it stands now, this test would pass if no data files were loaded, or if a subset of the data files in the input were loaded (and others were not).
I think the test should also verify that a `metadata.json` file is present, as right now the test would pass even if it were not. Ideally, the test should verify that the metadata.json contains the expected metadata based on the input.